spring-cloud-dataflow-admin-starter/src/main/java/org/springframework/cloud/dataflow/admin/controller/StreamDeploymentController.java

```diff
@@ -274,10 +274,20 @@ else if (hasOutput) {
 		for (Map.Entry<String, String> entry : streamDeploymentProperties.entrySet()) {
 			if (entry.getKey().startsWith(modulePrefix)) {
 				if (entry.getKey().startsWith(producerPropertyPrefix)) {
+					// Set per-binding properties
+					moduleDeploymentProperties.put(BindingProperties.OUTPUT_BINDING_KEY_PREFIX +
+							entry.getKey().substring(producerPropertyPrefix.length()), entry.getValue());
+					// Set ChannelBindingServiceProperties' producer properties
+					// todo: Once the producer/consumer properties are moved to per-binding properties, following step
```

Comment:
The properties are already per-binding properties, e.g.:

```
private int partitionCount = 1;

private String partitionKeyExpression;

private String partitionKeyExtractorClass;

private String partitionSelectorClass;

private String partitionSelectorExpression;
```

The challenge is that they are not qualified as "producer.*" properties at the binding level, nor should they be since each binding is already corresponding to a single direction (inbound vs. outbound).